src/musicbrainz/web/musicbrainzrecordingstask.cpp

```diff
@@ -167,6 +167,14 @@ void MusicBrainzRecordingsTask::doNetworkReplyFinished(
     // Continue with next recording id
     DEBUG_ASSERT(!m_queuedRecordingIds.isEmpty());
     emit currentRecordingFetched();
+    // Some of the tracks has loads of RecordingIds.
+    // This can cause hitting the rate limits of MusicBrainz.
+    // According to the MusicBrainz API Doc: https://musicbrainz.org/doc/MusicBrainz_API/Rate_Limiting
+    // The rate limit should be one query in a second.
+    // Related Bug: https://bugs.launchpad.net/mixxx/+bug/1983204
+    // In order to not hit the rate limits and respect their rate limiting rule.
+    // We are going to delay every request by one second.
+    QThread::msleep(1000);
```

Comment:
We can speed it up, by starting a timer before issue the first request and use here the remaining time to complete a second. This way 20 requests will take 20 s. Now it takes as long as before + 20 s.

Reply:
You may use https://github.com/mixxxdj/mixxx/blob/f2d670edeb483efb066bf7f00255d01df2f0017c/src/util/timer.h#L18
for that.

Reply:
I tried to use Timer class. I'm not sure if it is the best approach to check if it is not passed 1 seconds by using while loop.